spec.md

```diff
@@ -4695,17 +4698,40 @@ File = {Statement | newline} eof .
 
 Statement = DefStmt | IfStmt | ForStmt | SimpleStmt .
 
-DefStmt = 'def' identifier '(' [Parameters [',']] ')' ':' Suite .
+DefStmt = 'def' identifier '(' [Parameters [',']] ')' ['->' TypeExpr ]  ':' Suite .
 
 Parameters = Parameter {',' Parameter}.
 
-Parameter  = identifier
-           | identifier '=' Test
+Parameter  = identifier [':' TypeExpr] ['=' Test]
            | '*'
-           | '*' identifier
-           | '**' identifier
+           | '*' identifier [':' TypeExpr]
+           | '**' identifier [':' TypeExpr]
            .
 
+LambdaParameters = LambdaParameter {',' LambdaParameter}.
+
+LambdaParameter  = identifier ['=' Test]
+                 | '*'
+                 | '*' identifier
+                 | '**' identifier
+                 .
+           
+TypeExpr = Type {'|' Type}.
+
+Type = identifier [TypeArguments].
```

Comment:
Is there support for a syntactic sugaring of List[Int] or Tuple[String, Int] as [Int] and (String, Int) respectively?

Reply:
I thought Python had this but I can't find it now so probably not.

I'm tempted to say we should add it, but I fear it'd be useless if we (or users?) subtype these builtin types with ones that have more invariants, e.g. FrozenDict or whatnot.

Reply:
Yes that. Maybe you remember Buck having syntax sugar. They removed it. I think, because it was too confusing.